pkg/apis/networkoperator/v1/types.go

```diff
@@ -119,6 +123,59 @@ type OpenshiftSDNConfig struct {
 	UseExternalOpenvswitch *bool `json:"useExternalOpenvswitch,omitempty"`
 }
 
+type OpenStackCredentials struct {
+	// TODO(dulek): Might want to add all the options, e.g. keys.
+	AuthURL           string `json:"authURL"`
+	UserDomainName    string `json:"userDomainName"`
+	Username          string `json:"username"`
+	Password          string `json:"password"`
+	ProjectDomainName string `json:"projectDomainName"`
+	ProjectId         string `json:"projectId"`
+}
+
+// KuryrConfig configures the Kuryr-Kubernetes SDN
+type KuryrConfig struct {
+	// The base64-encoded CA certificate to use when connecting to OpenStack.
+	// +optional
+	OpenStackCACertificate string `json:"openStackCACertificate"`
+
+	OpenStackCredentials *OpenStackCredentials `json:"openStackCredentials"`
+
+	// If true, kuryr-daemon readiness and liveness probes are enabled.
+	// +optional
+	DaemonEnableProbes *bool `json:"daemonEnableProbes"`
+
+	// The port kuryr-daemon will listen for readiness and liveness requests.
+	// +optional
+	DaemonProbesPort *uint16 `json:"daemonProbesPort"`
+
+	// The CNI bin directory on the hosts, usually /opt/cni/bin.
+	CNIBinDir string `json:"cniBinDir"`
+
+	// If true, kuryr-controller readiness and liveness probes are enabled.
+	// +optional
+	ControllerEnableProbes *bool `json:"controllerEnableProbes"`
```

Comment:
Is there any reason for probes to be disabled?

Reply:
We need it as an option because probes support it depends on Kuryr version being used. We could switch to some autodetection, but let's start with that.

Reply:
Okay, seems like we won't support that old Kuryr, so whatever.